### PR TITLE
Unit tests for display.py

### PR DESCRIPTION
**Pull request by tkhanova**
Unit tests for display.py

### CODE REVIEW

**Comment by tkhanova** [comment]
@vasily-v-ryabov, could you please take a look?

File: test/test_xlib_display.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        self.assertEqual(self.display.get_pointer_mapping(), [0] * length)
+
+    def test_can_close_display(self):
+        self.display.close()
+        self.assertEqual(str(self.display.display.socket_error), "Display connection closed by client")
+
+    def test_can_close_display_and_check_for_error(self):
+        self.display.close()
+        with self.assertRaises(Xlib.error.ConnectionClosedError, None) as e:
+            self.display.flush()
```
Comment (vasily-v-ryabov) [inline test/test_xlib_display.py]: Context manager (`with` statement) is not implemented for `assertRaises` in Python 2.6. `self.assertRaises(Xlib.error.ConnectionClosedError, self.display.flush)` should work for both py2.6 and py2.7. Second argument of `assertRaises` is always callable object, not a statement.

File: test/test_xlib_display.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        self.assertNotEqual(fonts, [])
+
+    def test_lookup_valid_keysym(self):
+        self.assertNotEqual(self.display.lookup_string(65535L), None)
+
+    def test_lookup_invalid_keysym(self):
+        self.assertEqual(self.display.lookup_string(-1), None)
+
+    def test_rebind_string(self):
+        self.display.rebind_string(65535L, "qqq")
```
Comment (vasily-v-ryabov) [inline test/test_xlib_display.py]: `L` for integers is not supported in Py3.x. That's why Py3 tests are broken (syntax error in `test_xlib_display.py`). `__import__` call just doesn't output any details. But you may try to execute `test_xlib_display.py` only (not touching `testall.py`) and see which errors are here for Py3.

File: .checkignore [hunk 3]
```diff
@@ -0,0 +1 @@
+test/*
```
Comment (vasily-v-ryabov) [inline .checkignore]: Please remove `.checkignore`. We don't need to exclude test files from static analysis.

Comment (tkhanova) [inline .checkignore]: Is there any way to disable warnings for Test classes? QC complains about comments for each test.

**Comment by tkhanova** [comment]
@vasily-v-ryabov, I fixed all syntax errors for Python3, but there are other erros in all tests.

File: Xlib/protocol/display.py [hunk 4]
```diff
@@ -532,7 +532,7 @@ def send_and_recv(self, flush = None, event = None, request = None, recv = None)
             if ws:
                 try:
                     i = self.socket.send(self.data_send)
-                except socket.error, err:
+                except (socket.error, err) as err:
```
Comment (vasily-v-ryabov) [inline Xlib/protocol/display.py]: 2to3 fixer suggests to make it `except socket.error as err:`. Comma doesn't mean a list in Python 2 `except` statement.

File: Xlib/xobject/drawable.py [hunk 5]
```diff
@@ -229,7 +229,7 @@ def put_pil_image(self, gc, x, y, image, onerror = None):
             unit = self.display.info.bitmap_format_scanline_unit
             stride = roundup(width * unit, pad) >> 3
         else:
-            raise ValueError, 'Unknown data format'
+            raise (ValueError, 'Unknown data format')
```
Comment (vasily-v-ryabov) [inline Xlib/xobject/drawable.py]: Must be `raise ValueError('Unknown data format')`

**Comment by vasily-v-ryabov** [comment]
Could you merge recent fixes from @SvetlanaFilicheva (in upstream repo) to your fork? Maybe some fixes wouldn't be reverted for Py3 compatibility?

**Comment by tkhanova** [comment]
@vasily-v-ryabov, works like a charm.

**Comment by tkhanova** [comment]
```
mod = __import__(modname, globals())
ImportError: No module named 'unix_connect'
```
@vasily-v-ryabov , do you know how to fix it? 
fromlist=["."] option doesn't help.

**Comment by vasily-v-ryabov** [comment]
@tkhanova, no ideas yet.

**Comment by tkhanova** [comment]
fixed more issues, stuck with apply function

```
self._binary = apply(self._request.to_binary, args, keys)
NameError: global name 'apply' is not defined
```

**Comment by vasily-v-ryabov** [comment]
Could be replaced with `self._binary = self._request.to_binary(*args, **keys)`

**Comment by vasily-v-ryabov** [comment]
Unit tests for `display.py` are working on Py2, so merging it. Please open another PR if you could fix Py3 tests.

**Pull request merged**